### PR TITLE
refactor pr-check

### PR DESCRIPTION
**Pull request by erwinmombay**
Fixes https://github.com/ampproject/amphtml/issues/5848

on pull request:
if only markdown, skip everything
if html, run runtime tests
if starts with build-system/, run node/ava tests

on push or branch:
always run everything

### CODE REVIEW

File: build-system/pr-check.js [hunk 1]
```diff
@@ -113,14 +126,58 @@ function determineBuildTargets(filePaths) {
...
+
 /**
  * The main method for the script execution which much like a C main function
  * receives the command line arguments and returns an exit status.
  * @param {!Array<string>} argv
  * @returns {number}
  */
 function main(argv) {
-  const travisCommitRange = argv[2] || '';
+  const travisCommitRange = argv.range || '';
```
Comment (jridgewell) [inline build-system/pr-check.js]: While we're at this, let's fix this to be `master...${branch}`.

Comment (erwinmombay) [inline build-system/pr-check.js]: yep, thanks for the reminder!

File: build-system/pr-check.js [hunk 2]
```diff
@@ -91,37 +102,107 @@ function isValidatorFile(filePath) {
...
+    // the css files into js files.
+    execOrDie(`${gulp} dep-check`);
+    // Unit tests with Travis' default chromium
+    execOrDie(`${gulp} test --nobuild --compiled`);
+    // Integration tests with all saucelabs browsers
+    execOrDie(`${gulp} test --nobuild --saucelabs --integration --compiled`);
+    // All unit tests with an old chrome (best we can do right now to pass tests
+    // and not start relying on new features).
+    // Disabled because it regressed. Better to run the other saucelabs tests.
+    execOrDie(`${gulp} test --nobuild --saucelabs --oldchrome --compiled`);
```
Comment (erwinmombay) [inline build-system/pr-check.js]: @aghassemi double check your changes are here

File: build-system/pr-check.js [hunk 3]
```diff
@@ -91,37 +102,107 @@ function isValidatorFile(filePath) {
...
  * The main method for the script execution which much like a C main function
  * receives the command line arguments and returns an exit status.
  * @param {!Array<string>} argv
  * @returns {number}
  */
 function main(argv) {
-  const travisCommitRange = argv[2] || '';
-  const buildTargets = determineBuildTargets(filesInPr(travisCommitRange));
+  // If $TRAVIS_PULL_REQUEST_SHA is empty then it is a push build and not a PR.
+  if (!process.env.TRAVIS_PULL_REQUEST_SHA) {
```
Comment (jridgewell) [inline build-system/pr-check.js]: No idea.

File: build-system/pr-check.js [hunk 4]
```diff
@@ -91,37 +102,106 @@ function isValidatorFile(filePath) {
...
+    console.log('Running all commands on push build.');
+    runAllCommands();
+    return 0;
+  }
+  const travisCommitRange = `master...${process.env.TRAVIS_PULL_REQUEST_SHA}`;
+  const files = filesInPr(travisCommitRange);
+  const buildTargets = determineBuildTargets(files);
+
+  if (buildTargets.length == 1 && buildTargets[0] == 'DOCS') {
```
Comment (jridgewell) [inline build-system/pr-check.js]: `buildTargets` is a `Set`, so this will never be true. `buildTargets.has('DOCS')`

Comment (erwinmombay) [inline build-system/pr-check.js]: fixed

**Comment by erwinmombay** [comment]
@jridgewell will do. I'll revert ASAP if necessary

**Pull request merged**